kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateCodeUtil.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.maven.plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.artifact.resolver.filter.CumulativeScopeArtifactFilter;
+import org.apache.maven.project.MavenProject;
+import org.kie.memorycompiler.JavaCompilerSettings;
+import org.kie.memorycompiler.JavaConfiguration;
+import org.kie.memorycompiler.KieMemoryCompiler;
+
+public class GenerateCodeUtil {
+
+    public static URLClassLoader getProjectClassLoader(MavenProject project, File outputDirectory, JavaCompilerSettings javaCompilerSettings) {
+        try {
+            Set<URL> urls = new HashSet<>();
+            for (String element : project.getCompileClasspathElements()) {
+                File file = new File(element);
+                javaCompilerSettings.addClasspath(file);
+                urls.add(file.toURI().toURL());
+            }
+
+            project.setArtifactFilter(new CumulativeScopeArtifactFilter(Arrays.asList("compile", "runtime")));
+            for (Artifact artifact : project.getArtifacts()) {
+                File file = artifact.getFile();
+                if (file != null) {
+                    javaCompilerSettings.addClasspath(file);
+                    urls.add(file.toURI().toURL());
+                }
+            }
+            urls.add(outputDirectory.toURI().toURL());
+
+            return URLClassLoader.newInstance(urls.toArray(new URL[0]), GenerateCodeUtil.class.getClassLoader());
+
+        } catch (DependencyResolutionRequiredException | MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void compileAndWriteClasses(File targetDirectory, ClassLoader projectClassLoader, JavaCompilerSettings javaCompilerSettings, Map<String, String> classNameSourceMap, boolean dumpGeneratedSources) {
+        if (dumpGeneratedSources) {
+            dumpGeneratedSources(targetDirectory, classNameSourceMap);
+        }
+
+        Map<String, byte[]> compiledClassesMap = KieMemoryCompiler.compileNoLoad(classNameSourceMap, projectClassLoader, javaCompilerSettings, JavaConfiguration.CompilerType.ECLIPSE);
+
+        for (Map.Entry<String, byte[]> entry : compiledClassesMap.entrySet()) {
+            Path packagesDestinationPath = Paths.get(targetDirectory.getPath(), "classes", entry.getKey().replace('.', '/') + ".class");
```

Comment:
hi @mariofusco 
Could you please replace "/" with OS-specific path separator, all over this class ?  Thanks!